Gems/ROS2/Code/Source/SimulationUtils/FollowingCameraComponent.cpp

```diff
@@ -0,0 +1,230 @@
+                        "Rotation Speed",
+                        "Rotation Speed around the target")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &FollowingCameraComponent::m_predefinedViews, "Views", "Views to follow")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &FollowingCameraComponent::m_defaultView, "Default View", "Default View to follow")
+                    ->Attribute(AZ::Edit::Attributes::Min, 0);
+            }
+        }
+    }
+
+    void FollowingCameraComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC("TransformService"));
+        required.push_back(AZ_CRC("CameraService"));
+    }
+
+    void FollowingCameraComponent::Activate()
+    {
+        if (m_predefinedViews.size() == 0)
+        {
+                AZ_Warning("FollowingCameraComponent", false, "No predefined views");
+                return;
+        }
+        if (m_defaultView < m_predefinedViews.size())
+        {
+            m_currentView = m_predefinedViews[m_defaultView];
+        }
+        InputChannelEventListener::Connect();
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void FollowingCameraComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+        InputChannelEventListener::Disconnect();
+    }
+
+    void FollowingCameraComponent::CacheTransform(const AZ::Transform& transform, float deltaTime)
+    {
+        // update the smoothing buffer
+        m_lastTranslationsBuffer.push_back(AZStd::make_pair(transform.GetTranslation(), deltaTime));
+        m_lastRotationsBuffer.push_back(AZStd::make_pair(transform.GetRotation(), deltaTime));
+
+        if (m_lastTranslationsBuffer.size() > m_smoothingBuffer)
+        {
+            m_lastTranslationsBuffer.pop_front();
+        }
+        if (m_lastRotationsBuffer.size() > m_smoothingBuffer)
+        {
+            m_lastRotationsBuffer.pop_front();
+        }
+    }
+    void FollowingCameraComponent::OnTick(float deltaTime, AZ::ScriptTimePoint /*time*/)
+    {
+        AZ_Warning("FollowingCameraComponent", m_currentView.IsValid(), "View is not valid");
+        if (!m_currentView.IsValid())
+        {
+            return;
+        }
+        // obtain the current view transform
+        AZ::Transform target_local_transform;
+        AZ::Transform target_world_transform;
+
+        AZ::TransformBus::Event(m_currentView, &AZ::TransformBus::Events::GetLocalAndWorld, target_local_transform, target_world_transform);
+
+        // get parent's transform
+        const AZ::Transform parent_transform = target_world_transform * target_local_transform.GetInverse();
+
+        CacheTransform(parent_transform, deltaTime);
+
+        // get the averaged translation and quaternion
+        AZ::Transform filtered_parent_transform = { SmoothTranslation(), SmoothRotation(), 1.f };
+
+        // modifier for zooming
```

Comment:
Nit: this comment does not help much. Either remove or expand, right now it is captured in variable name.